src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -13,8 +13,14 @@ namespace AutoRest.Swagger.Model.Utilities
 {
     public static class ValidationUtilities
     {
-        private static readonly string XmsPageable = "x-ms-pageable";
+        private const string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private const string AzureHostSignature = "management.azure.com";
+
+        // if x-ms-parameterized-host extension is used or if the host is anything of the form management.azure, assume this to be an ARMServiceDefinition
+        // TODO: refine this definition for azure data plain also check x-ms-parameterized-host in detail to ensure this is ARM
+        public static bool IsARMServiceDefinition(ServiceDefinition serviceDefinition) => 
```

Comment:
Agreed but we wanted to isolate this validation for ARM specs only, it shouldn't throw validation errors for something like PetStore
The effort could be useful in a couple of other validation rule implementations where we also want to enforce url path patterns but for ARM specs only.
I also feel in the long run this differentiation could help us create something like a base validator that can validate the swagger spec irrespective of whether it is ARM or not and a separate ARM-specific validator with all the complex details